### PR TITLE
docs: Add AFM section

### PR DESCRIPTION
**Pull request by manzt**

### CODE REVIEW

File: docs/src/pages/en/anywidget-front-end-module.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+dependencies](https://github.com/marimo-team/marimo/blob/7f3023ff0caef22b2bf4c1b5a18ad1899bd40fa3/frontend/src/plugins/impl/anywidget/AnyWidgetPlugin.tsx#L161-L267).
+
+## Framework Bridges
+
+AFM intentionally does not prescribe specific models for state management or UI
+rendering. While many front-end tools exist to help with authoring UIs (e.g.,
+React, Svelte, Vue) we strongly believe that incorporating these
+non-web-standard pieces at the specification level would be a mistake. Our goal
+is to create a solution for reusable widgets that aligns with the web's strong
+backwards compatibility garantuees.
```
Comment (gereleth) [inline docs/src/pages/en/anywidget-front-end-module.md]: backwards compatibility ~~garantuees~~guarantees.

File: docs/src/pages/afm.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+title: "Anywidget Front-End Module (AFM)"
+description: "A specification for portable widgets based on ECMAScript modules."
+layout: ../../layouts/MainLayout.astro
+---
+
+## What is AFM?
+
+The **Anywidget Front-End Module (AFM)** specification defines a standard for
+creating portable widget front-end code. Our vision is to enable widget reuse
+**beyond Jupyter**, including other computational notebooks and standalone web
```
Comment (nvictus) [inline docs/src/pages/afm.md:11]:
```suggestion
within and **beyond Jupyter**, including other computational notebooks and standalone web
```

File: docs/src/pages/afm.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+### Host platform
+
+The web-based environment in which a widget is embedded. It is responsible for
+loading AFM and calling [lifecycle methods](#lifecycle-methods) with required
+the platform APIs.
+
+The `anywidget` Python library provides the necessary glue code to make all
+Jupyter-like environments (e.g., Jupyter Notebook, JupyterLab, Google Colab, VS
+Code) an AFM-compatible host platform. The
+[marimo](https://github.com/marimo-team/marimo) project is a _native_ host
```
Comment (nvictus) [inline docs/src/pages/afm.md:38]:
```suggestion
[marimo](https://github.com/marimo-team/marimo) project is an example of a _native_ host
```

File: docs/src/pages/afm.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+- `initialize({ model })`: Executed once per widget instance during model
+initialization. It has access to `model` to setup non-view event handlers or
+state to share across views.
+
+- `render({ model, el })`: Executed once per view during view rendering. It has
+access to both the `model` and an `el` DOM element. Used to setup event
+handlers or access state specific to that view.
+
+Each method MAY return a **cleanup function** which will be called when the
```
Comment (nvictus) [inline docs/src/pages/afm.md:97]: Does the cleanup function have a signature (take optional arguments)?

File: docs/src/pages/afm.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+dependencies](https://github.com/marimo-team/marimo/blob/7f3023ff0caef22b2bf4c1b5a18ad1899bd40fa3/frontend/src/plugins/impl/anywidget/AnyWidgetPlugin.tsx#L161-L267).
+
+## Framework Bridges
+
+AFM intentionally does not prescribe specific models for state management or UI
+rendering. While many front-end tools exist to help with authoring UIs (e.g.,
+React, Svelte, Vue) we strongly believe that incorporating these
+non-web-standard pieces at the specification level would be a mistake. Our goal
+is to create a solution for reusable widgets that aligns with the web's strong
+backwards compatibility garantuees.
```
Comment (nvictus) [inline docs/src/pages/afm.md:190]:
```suggestion
backwards compatibility guarantees.
```

File: docs/src/pages/afm.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+  }
+}
+```
+
+A host platform is expected to:
+
+- Load this module.
+- Call the lifecycle methods, passing in dependencies (`model` & `el`).
+
+All browsers support ESM, so loading the module is supported natively across
```
Comment (nvictus) [inline docs/src/pages/afm.md:70]:
```suggestion
All browsers support ECMAScript modules, so loading the module is supported natively across
```

File: docs/src/pages/afm.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+layout: ../../layouts/MainLayout.astro
+---
+
+## What is AFM?
+
+The **Anywidget Front-End Module (AFM)** specification defines a standard for
+creating portable widget front-end code. Our vision is to enable widget reuse
+**beyond Jupyter**, including other computational notebooks and standalone web
+applications. AFM is oriented around a minimal set of APIs we identified as
+essential for integration with _host platforms_, boiling down to:
```
Comment (nvictus) [inline docs/src/pages/afm.md:13]:
```suggestion
essential for integration with [_host platforms_](#host-platform), boiling down to:
```

**Review by nvictus — APPROVED** [review]

File: docs/src/pages/afm.md (hunk 4 above)
Comment (manzt) [reply docs/src/pages/afm.md:97]: Nope, just a () => void. You can use a closure to capture any deps.

File: docs/src/pages/en/anywidget-front-end-module.md (hunk 1 above)
Comment (manzt) [reply docs/src/pages/en/anywidget-front-end-module.md]: Thanks!!

**Comment by jtpio** [comment]
Landing here after the lightning talk by @MarcSkovMadsen at PyData Paris 2024 :)

Just wanted to say having some kind of "standard" for portable widget code would indeed be really useful :+1: 

For reference, a similar idea was discussed a few years ago at the Jupyter Widgets community workshop in London (https://github.com/jupyter-widgets/team-compass/issues/16), which led to some experimentation in https://github.com/jupyterlab/richoutput-js.

**Pull request merged**